### PR TITLE
docs/ci: Update config validation

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: docs/ci: Update config validation
Additional Description:

- some config snippets in docs were separated from `.rst` files, some of these can be checked with config validation (with some small modifications). Some fail - so will follow up in subsequent PR to address failing configs
- use glob/exclusions for matching config files in `examples/BUILD` 
- remove `verify_build_configs` as configs are found dynamically
- cleanup indentation on docs configs

Risk Level: low
Testing: yep
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: docs/BUILD [hunk 1]
```diff
@@ -1,3 +1,27 @@
...
+envoy_package()
+
+# dns-cache-circuit-breaker fails on windows: "Error: unable to read file: /etc/ssl/certs/ca-certificates.crt"
+
+filegroup(
+    name = "configs",
+    srcs = glob(
+        ["root/**/*.yaml"],
+        exclude = [
+            "root/intro/*",
```
Comment (phlax) [inline docs/BUILD:19]: remove

**Comment by phlax** [comment]
docs are here:

https://387472-65214191-gh.circle-artifacts.com/0/generated/docs/index.html

File: docs/BUILD [hunk 2]
```diff
@@ -1,3 +1,26 @@
...
+filegroup(
+    name = "configs",
+    srcs = glob(
+        ["root/**/*.yaml"],
+        exclude = [
+            "root/intro/_include/life-of-a-request.yaml",
+            "root/intro/arch_overview/security/_include/ssl.yaml",
+            "root/configuration/http/http_filters/_include/dns-cache-circuit-breaker.yaml",
+            "root/configuration/http/http_filters/_include/grpc-reverse-bridge-filter.yaml",
+            "root/configuration/http/http_filters/_include/grpc-transcoder-filter.yaml",
```
Comment (mattklein123) [inline docs/BUILD:23]: Why are these excluded? Similar to the other one? Can you add a comment/TODO?

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Amazing, thanks. Just one small question/comment.

/wait

File: docs/BUILD (hunk 2 above)
Comment (phlax) [reply docs/BUILD:23]: they fail on everything - probs deprecated configs iirc - ill add todo to fix them...

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**